src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var list = 1;
+
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var list = 1;
+
+        var {|Rename:list1|} = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < list1.Length; i++)
+        {
+            var a = list1[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task IndexConflict()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array)
+        {
+            int i = 0;
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i1|} = 0; i1 < array.Length; i1++)
+        {
+            var a = array[i1];
+            int i = 0;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task VariableWritten()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array)
+        {
+            a = 1;
+        }
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task WrongCaretPosition()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach (var a in array)
+        {
+            [||] 
+        }
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CaretPosition1()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        [||] foreach (var a in array)
+        {
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++)
+        {
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CaretPosition2()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach (var a in array) [||] 
```

Comment:
also seems undesirable.

Reply:
same here.

Reply:
Same reason as above.  We did a lot of work to narrow down all our refactorings that were providing too large a span to be activated on.

Reply:
tracking here - https://github.com/dotnet/roslyn/issues/25667